### DISCUSSION TITLE
Numbers in csv displayed as text instead of numbers

### DISCUSSION THREAD

**Discussion by saginadav** [PHPOffice/PhpSpreadsheet#2629] [Q&A] [answered]
Hey All I have this question
I created a spreadsheet and inserted numbers, the problem is that in the csv it says it is text. and not recognized as number, Therefor the functions not worked (for example the sum method in the xls).
I have this code that runs on each cell and change it to number 
`$sheet->getStyle($cell)->getNumberFormat()->setFormatCode($format[0]);`
when I am in the xls, and stand on the cell it says in the top that this is a number, but get message that this is a text.
how I can tell to the xls this is a number?
Since now the xls is useless, it is just like an image, I need it to be numbers that users can summarized columns 
![image](https://user-images.githubusercontent.com/18045448/155882685-c8efcb45-bd92-4f2a-85e7-87f73a937e71.png)

**Comment by MarkBaker** [comment]
How did you insert those numbers? Did you load them from a csv file? (Your descriptions is rather vague). If so, does the csv have vales containing commas like `1,572,637` (possibly wrapped in quotes)? If so, then it isn't a number; it's a string. You'll need to iterate over those cells and change the value manually to a numeric value by stripping the commas, and casting to `int` or `float`.

Setting a format code for a cell using
```php
$sheet->getStyle($cell)->getNumberFormat()->setFormatCode($format[0]);
```
doesn't change the content of that cell. It won't magically convert a string  value into a number. It simply tells Excel how to format that value for display purposes; but if the value is a string rather than a numeric, then it won't change anything.
When Excel tells you it's a number, it doesn't tell you that; it tells you that you've tried to format it as a number, not that it is a number.

**Comment by saginadav** [comment, marked as answer]
the solution is first determine the cell as numeric and that set the formatCode.
```
$dataType[0] = DataType::TYPE_NUMERIC;
$format[0] = \PhpOffice\PhpSpreadsheet\Style\NumberFormat::FORMAT_PERCENTAGE_00;
$sheet->setCellValueExplicit($cell,$innervalue, $dataType[0]);
$sheet->getStyle($cell)->getNumberFormat()->setFormatCode($format[0]);
```
